### PR TITLE
feat: add quickfix to derive missing type classes

### PR DESCRIPTION
**Pull request by sockmaster27**
Proposes quickfix to derive type classes (`Eq`, `Order`, `ToString`) for enums when trying to do something that depends on them.

### CODE REVIEW

**Comment by sockmaster27** [comment]
This does not include restrictable enums because I don't know what they are

**Comment by magnus-madsen** [comment]
> This does not include restrictable enums because I don't know what they are

No worries, they are experimental anyway.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala [hunk 1]
```diff
@@ -230,6 +237,44 @@ object CodeActionProvider {
...
+    * Returns a quickfix code action to derive the `ToString` type class for the given type `tpe` if it is an enum.
+    */
+  private def mkDeriveMissingToString(tpe: Type, uri: String): Option[CodeAction] =
+    mkDeriveMissing(tpe, "ToString", uri)
+
+  /**
+    * Internal helper function for all `mkDeriveMissingX`.
+    * Returns a quickfix code action to derive the given type class `clazz` for the given type `tpe` if it is an enum.
+    */
+  private def mkDeriveMissing(tpe: Type, clazz: String, uri: String): Option[CodeAction] = tpe.typeConstructor match {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala:262]: Any particular reason not to reuse `mkDerive`?

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala (hunk 1 above)
Comment (sockmaster27) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala:262]: Just that one is a refactor and one is a quickfix.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**